compose/internal/service/import_session.go

```diff
@@ -0,0 +1,118 @@
+package service
+
+import (
+	"context"
+	"time"
+
+	"github.com/cortezaproject/corteza-server/internal/auth"
+	"github.com/titpetric/factory"
+
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+)
+
+type (
+	recordSet []*RecordImportSession
+
+	importSession struct {
+		logger  *zap.Logger
+		records recordSet
+	}
+
+	ImportSessionService interface {
+		FindRecordByID(ctx context.Context, sessionID uint64) (*RecordImportSession, error)
+		SetRecordByID(ctx context.Context, sessionID, namespaceID, moduleID uint64, fields map[string]string, progress *RecordImportProgress, decoder *Decoder) (*RecordImportSession, error)
+		DeleteRecordByID(ctx context.Context, sessionID uint64) error
+	}
+)
+
+func ImportSession() ImportSessionService {
+	return (&importSession{
+		logger:  DefaultLogger.Named("importSession"),
+		records: recordSet{},
+	})
+}
+
+func (svc importSession) indexOf(userID, sessionID uint64) int {
+	for i, r := range svc.records {
+		if r.SessionID == sessionID && r.UserID == userID {
+			return i
+		}
+	}
+
+	return -1
+}
+
+func (svc *importSession) FindRecordByID(ctx context.Context, sessionID uint64) (*RecordImportSession, error) {
+	userID := auth.GetIdentityFromContext(ctx).Identity()
+	i := svc.indexOf(userID, sessionID)
+	if i >= 0 {
+		return svc.records[i], nil
+	}
+	return nil, errors.New("Can't access session: session not found")
+}
+
+func (svc *importSession) SetRecordByID(ctx context.Context, sessionID, namespaceID, moduleID uint64, fields map[string]string, progress *RecordImportProgress, decoder *Decoder) (*RecordImportSession, error) {
+	userID := auth.GetIdentityFromContext(ctx).Identity()
+	i := svc.indexOf(userID, sessionID)
+	var ris *RecordImportSession
+
+	if i >= 0 {
+		ris = svc.records[i]
+	} else {
+		ris = &RecordImportSession{
+			SessionID: factory.Sonyflake.NextID(),
+			CreatedAt: time.Now(),
+		}
+		svc.records = append(svc.records, ris)
```

Comment:
This (modifying same variable from multiple goroutines) is not safe.
Add locking (see internal/permissions/service.go for examples)